src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -978,6 +981,91 @@ export class ResultsViewPageStore {
+        const samples = this.samples.result!;
+        const patients = this.patients.result!;
+
+        if (this.oqlQuery.trim() === '') {
+            return Promise.resolve([]);
+        } else {
+            const filteredAlterationsByOQLLine: UnflattenedOQLLineFilterOutput<AnnotatedExtendedAlteration>[] = (
+                filterCBioPortalWebServiceDataByUnflattenedOQLLine(
+                    this.oqlQuery,
+                    data,
+                    accessorsInstance,
+                    defaultOQLQuery
+                )
+            );
+
+            return Promise.resolve(filteredAlterationsByOQLLine.map(
+                (oql) => ({
+                    cases: groupDataByCase(oql, samples, patients),
+                    oql,
+                    list: filterSubQueryData(
+                        oql, defaultOQLQuery,
+                        data, accessorsInstance,
+                        samples, patients
+                    )
+                })
+            ));
+        }
+    }
+});
+
+    readonly isSampleAlteredMap = remoteData({
+        await: () => [this.filteredAlterationsDataByUnflattenedOQLLine, this.samples],
+        invoke: async() => {          
+            const result : {[x: string]: boolean[]} = {};
+            this.filteredAlterationsDataByUnflattenedOQLLine.result!.forEach((element, key) => {
+                //1: is not group(1.1, 1.2)
+                if (element.list === undefined) {
+                    const notGroupedOql = element.oql as OQLLineFilterOutput<AnnotatedExtendedAlteration>;                    
+                    const sampleKeys = _.map(notGroupedOql.data, (data) => data.uniqueSampleKey);
+                    const isOnlyGene = (oql_parser.parse(this.oqlQuery)![key] as oql_parser.SingleGeneQuery).alterations === false;
+                    //1.1: gene
+                    if (isOnlyGene) {
+                        result[notGroupedOql.gene] = this.samples.result.map((sample: Sample) => {
+                            return _.includes(sampleKeys, sample.uniqueSampleKey);
+                        });
+                    }
+                    //1.2: gene alteration
+                    else {
+                        result[notGroupedOql.oql_line.slice(0, -1)] = this.samples.result.map((sample: Sample) => {
+                            return _.includes(sampleKeys, sample.uniqueSampleKey);
+                        });
+                    }
+                }
+                //2: is group
+                else {
+                    const groupedOql = element.oql as MergedTrackLineFilterOutput<AnnotatedExtendedAlteration>;
+                    const oqlData = _.flatten(_.map(groupedOql.list, (list) => list.data));
+                    const sampleKeys = _.map(oqlData, (data) => data.uniqueSampleKey);
+                    const resultKey = groupedOql.label! === undefined ? _.map(groupedOql.list, (data) => data.gene).join(' / ') : groupedOql.label!;
```

Comment:
i don't think .label! makes sense because the ! = NOT undefined, but you're testing here whether it's undefined or not